numpy_turtle/numpy_turtle.py

```diff
@@ -134,8 +133,10 @@ def push(self):
 
     def pop(self):
         """Restore the state that was last pushed.
+           allow for handling of pops of empty stacks
```

Comment:
I think that by default, this shouldn't be allowed. However, I like the idea of allowing this by setting a boolean keyword argument, so that the method signature becomes something like `def pop(self, *, allow_empty=False)`, and when called with `.pop(allow_empty=True)`, it would allow the stack to be empty. Otherwise, You could raise a `RuntimeError` with a message that the stack is empty and `allow_empty` can be used to pop anyway.